views/js/checkout-component-renderer.js

```diff
@@ -22,7 +22,341 @@
+        }
+    )
+
+    window.adyenCheckout = new AdyenCheckout(configuration);
+
+    // use this object to iterate through the stored payment methods
+    var paymentMethods = window.adyenCheckout.paymentMethodsResponse.paymentMethods;
+
+    // Iterate through the payment methods list we got from the adyen checkout component
+    paymentMethods.forEach(function (paymentMethod) {
+        //  if the container doesn't exits don't try to render the component
+        var paymentMethodContainer = $('[data-local-payment-method="' + paymentMethod.type + '"]');
+
+        // container doesn't exist, something went wrong on the template side
+        if (!paymentMethodContainer.length) {
+            return;
+        }
+
+        /* Subscribes to the adyen payment method form submission */
+        var paymentForm = $(".adyen-payment-form-" + paymentMethod.type);
+
+        var component = renderPaymentComponent(paymentMethod, paymentMethodContainer);
+
+        // Use data to reteive the payment method data
+        subscribeToPaymentFormSubmit(paymentForm, paymentMethod, component);
+    });
+
+    var checkoutStoredPaymentMethods = window.adyenCheckout.paymentMethodsResponse.storedPaymentMethods;
+
+    // Iterate through the stored payment methods list we got from the adyen checkout component
+    checkoutStoredPaymentMethods.forEach(function (storedPaymentMethod) {
+
+        //  storedPaymentMethod.id = $storedPaymentApiId in the stored-payment-method.tpl
+        //  don't try to render the component if the container doesn't exist
+        var paymentMethodContainer = $('[data-stored-payment-api-id="' + storedPaymentMethod.id + '"]');
+
+        // container doesn't exist, something went wrong on the template side
+        if (!paymentMethodContainer.length) {
+            return;
+        }
+
+        var component = renderPaymentComponent(storedPaymentMethod, paymentMethodContainer);
+
+        /* Subscribes to the adyen payment method form submission */
+        var paymentForm = $(".adyen-payment-form-" + storedPaymentMethod.id);
+
+        subscribeToPaymentFormSubmit(paymentForm, component);
+    });
+
+    function renderPaymentComponent(paymentMethod, paymentMethodContainer) {
+
+        if (notSupportedComponents.includes(paymentMethod.type)) {
+            return;
+        }
+
+        var configuration = Object.assign(paymentMethod, {
+            'onChange':handleOnChange
+        });
+
+        try {
+            return adyenCheckout
+            .create(paymentMethod.type, configuration)
+            .mount(paymentMethodContainer.find('[data-adyen-payment-container]').get(0));
+        } catch (err) {
+            console.log(paymentMethod.type);
+            console.log(err);
+            // The component does not exist yet
+        }
+    }
+
+    function subscribeToPaymentFormSubmit(paymentForm, paymentMethod, component) {
+        paymentForm.on('submit', function(e) {
+            e.preventDefault();
+
```

Comment:
Should we check if the PM is Adyen's before injecting our logic?

Reply:
Not necessary because we only loop through the Adyen payment methods list.